path_role.go

```diff
@@ -0,0 +1,433 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"regexp"
+	"strings"
+)
+
+// Constant to store the current role storage version
+var (
+	currentRoleStorageVersion = 1
+)
+
+// Constants for role specific data
+const (
+	DefaultTTLSeconds = 1800
+	MaxRoleNameLength = 50
+	// Increasing this above this limit might require implementing
+	// client-side paging in the filterGroupMembership API
+	MaxOCIDsPerRole = 100
+)
+
+func pathRole(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "role/" + framework.GenericNameRegex("role"),
+		Fields: map[string]*framework.FieldSchema{
+			"role": {
+				Type:        framework.TypeString,
+				Description: "Name of the role.",
+			},
+			"description": {
```

Comment:
Roles _typically_ don't have description fields.

Reply:
Have removed this field.